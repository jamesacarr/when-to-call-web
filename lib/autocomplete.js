let autocompleteService;

const getAutocompleteService = () => {
  if (!autocompleteService) {
    autocompleteService = new window.google.maps.places.AutocompleteService();
  }

  return autocompleteService;
};

export default input => {
  return new Promise(resolve => {
    if (!input) {
      resolve([]);
      return;
    }

    getAutocompleteService().getPlacePredictions({ input, types: ['establishment'] }, resolve);
  });
};
