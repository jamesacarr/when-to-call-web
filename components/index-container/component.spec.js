import React from 'react';
import { shallow } from 'enzyme';

import autocomplete from '../../lib/autocomplete';
import IndexContainer from './component';

jest.mock('../../lib/autocomplete.js', () => jest.fn(() => Promise.resolve([{ id: 'a' }, { id: 'b' }])));

describe('<IndexContainer/>', () => {
  describe('.render', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<IndexContainer><div>Test</div></IndexContainer>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('.search', () => {
    it('calls autocomplete', () => {
      const wrapper = shallow(<IndexContainer><div>Test</div></IndexContainer>);
      const instance = wrapper.instance();
      instance.search('test');

      expect(autocomplete).toHaveBeenCalledWith('test');
    });

    it('sets state when autocomplete resolve', async () => {
      const wrapper = shallow(<IndexContainer><div>Test</div></IndexContainer>);
      const instance = wrapper.instance();
      await instance.search('test');

      expect(wrapper.state('results')).toEqual([{ id: 'a' }, { id: 'b' }]);
    });
  });
});
