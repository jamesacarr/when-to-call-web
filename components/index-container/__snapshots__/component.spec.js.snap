// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexContainer/> .render renders correctly 1`] = `
<div>
  <WithStyles(TitleBar)
    onSearchChange={[Function]}
  />
  <WithStyles(ResultList)
    results={Array []}
  />
</div>
`;
